README.md

```diff
@@ -1,2 +1,152 @@
+* `dev_handler`: contains routines for managing the connections between the Raspberry Pi and the devices, the main routine for managing the managers of the connections, utilities for unpacking and packing messages to be sent to and from the devices, test files, and documentation.
+* `executor`: contains routines for spawning new processes to run the student code (written in Python) in the various modes (autonomous, teleop, coding challenge), an implementation of the StudentAPI (the functions that PiE provides for students to interact with the devices and gamepad) in Cython, the student code itself, test files, and documentation.
+* `shm_wrapper` and `shm_wrapper_aux`: contains the implementations of the tools described in the previous section, test files for both wrappers, a simple program in each folder to probe the state of the corresponding shared memory blocks while Runtime is running, and documentation.
+* `logger`: contains the implementation of the tool described in the previous section, a configuration file, test files, and documentation.
+* `runtime_util`: contains general utility functions that are useful to all of the processes, mostly dealing with getting information about device types and their parameters. The header file defines many important constants that are used throughout Runtime.
+* `systemd-stuff`: contains the services that are put into `systemd` that start Runtime on bootup of the Raspberry Pi.
+
+## Dependencies and Concepts
+
+Remember, one of the design principles is KISS. Nevertheless, Runtime is a complex system and will require effort to learn and understand. Hopefully this list does not seem too daunting to you! <sup id="return3">[3](#footnote3)</sup>
+
+Third-party library dependencies:
+
+* `Cython`: this library is used by `executor` to implement the Student API in a way that is both callable from Runtime (which is written in C) and from student code (which is written in Python)
+    * Documentation: https://cython.readthedocs.io/en/latest/
+* Google `protobuf` and `protobuf-c`: Google `protobuf` is the library that we use to serialize our messages between Runtime and Shepherd, and Runtime and Dawn. The brief explanation of how it works is this: the user defines the structure of a message in "protobuf language", and saves it as a `.proto` file. Google's protobuf compiler will then take that `.proto` file and generate code that can be used in a desired target language to serialize and deserialize ("pack" and "unpack" in the language of protobufs) messages of that type. Since Google's protobuf compiler does not have native support for C, we need to use the third party library `protobuf-c` to generate C code. (But `protobuf-c` makes use of Google's library, so we still need it).
+    * `protobuf` documentation: https://developers.google.com/protocol-buffers/docs/proto3
```

Comment:
Rename to `proto3` language documentation

Reply:
👌